pkg/scheduler/objects/queue.go

```diff
@@ -269,12 +269,14 @@ func (sq *Queue) setResources(resource configs.Resources) error {
 	} else {
 		log.Logger().Debug("max resources setting ignored: cannot set zero max resources")
 	}
+	sq.updateMaxResourceMetrics()
```

Comment:
I think we need this inside the `if` clause with the check for `StrictlyGreaterThanZero`. If there is no max set the object is a non nil pointer with no content.  We do not allow 0 values without at least 1 value larger than 0. That means this call would not result in any metrics being generated. The for loop inside the  `updateMaxResourceMetrics` never executes. It thus is a call to nowhere.

Same for guaranteed resources.

Reply:
I agree. Updated